fix: enhance terminal pixel detection and response handling

yorukot opened the pull request:
fix #900 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a development script to automate formatting, linting, testing, integration tests, and building the project.
* **Refactor**
  * Improved terminal cell size detection for file previews, providing more reliable and synchronous behavior across platforms.
* **Chores**
  * Updated terminal capability checks by removing Alacritty from the list of terminals assumed to support the Kitty graphics protocol.
  * Introduced a Makefile to streamline build, test, lint, and development workflows.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yorukot commented:
@coderabbitai full review

yorukot commented:
@lazysegtree Can you test if this successful fix the issue? Thanks

lazysegtree commented on format.sh:
```diff
@@ -0,0 +1,13 @@
+golangci-lint run
+
+# build the app
+CGO_ENABLED=0 go build -o ./bin/spf
```
The name of file is `format.sh`, but we are also building code ?

Should be format and build ?

Lets have a script that does all

Format
Runs unit tests
(Optionally) Runs testsuite
Builds spf

lazysegtree commented:
This is still not fixing the first character being ignored issue for me on Iterm2.

FYI, not having this issue on kitty.

yorukot commented:
I realized that my previous approach to detecting terminal cell size was actually quite inappropriate, it relied on terminal-specific escape sequences that many terminals either don't support or respond to inconsistently. As a result, terminal size detection failed on a wide range of terminals.

I've now replaced it with a much more reliable method using the ioctl system call (TIOCGWINSZ), which provides accurate pixel and character dimensions on almost all Unix-like terminals. This change greatly improves compatibility and correctness across environments like warp.

Additionally, I added a small development script dev.sh to streamline local testing and development workflows.

yorukot commented:
@coderabbitai summary

lazysegtree commented:
Works in Iterm2 now.

lazysegtree commented on dev.sh:
```diff
@@ -0,0 +1,215 @@
+
+    # Install requirements if needed (you might want to do this manually)
+    print_step "Installing testsuite requirements..."
+    if pip3 install -r requirements.txt > /dev/null 2>&1; then
```
@coderabbitai

This pollutes global python env. Its fairly common to use virtual envs, and not install packages in global python. 
So this might be unwanted for many users.

Please create a github issue that tracks adding functionility in dev.sh script to allow option to
have script create a venv in testsuite/ directory and use that to install packages and run testsuite.
